### PR TITLE
Add startPosition option for FileIO API

### PR DESCRIPTION
**Pull request by patriknw**
This is forward port of https://github.com/akka/akka/pull/22657
It was not trivial merge, so please review.

* New startPosition parameter to FileIO.toPath
* New startPosition parameter to FileIO.fromPath
* default startPosition value : 0
* fix binary compatibilities
* Add the MiMa filters for the internal classes
* Feedback from comments of the PR

### CODE REVIEW

**Comment by patriknw** [comment]
Refs #19483

File: akka-stream/src/main/scala/akka/stream/scaladsl/FileIO.scala [hunk 1]
```diff
@@ -86,5 +104,21 @@ object FileIO {
...
+   * and a possible exception if IO operation was not completed successfully.
+   *
+   * This source is backed by an Actor which will use the dedicated `akka.stream.blocking-io-dispatcher`,
+   * unless configured otherwise by using [[ActorAttributes]].
+   *
+   * @param f the file path to write to
+   * @param options File open options, defaults to Set(WRITE, CREATE)
+   * @param startPosition the start position to write to
+   */
+  def toPath(f: Path, options: Set[OpenOption], startPosition: Long): Sink[ByteString, Future[IOResult]] =
```
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/scaladsl/FileIO.scala]: good type here, with the fix already ( as in #22770  )

**Review by ktoso — APPROVED** [review]
LGTM, compared this and the original patch - looks good.

**Review by jrudolph — APPROVED** [review]
LGTM

**Pull request merged**